File: signal-matrix/python/signals/ai_reachable.py
# Copyright © 2026 Sthenos Security. All rights reserved.
"""REACHABLE: AI — called from entrypoint."""

def run_llm_query(user_prompt: str) -> str:
    """LLM01 REACHABLE: unsanitized user input to OpenAI."""
    from openai import OpenAI
    client = OpenAI()
    resp = client.chat.completions.create(
        model="gpt-4",
        messages=[{"role": "user", "content": user_prompt}]  # VIOLATION: unsanitized
    )
    return resp.choices[0].message.content

File: signal-matrix/javascript/signals/ai_reachable.js
'use strict';
// REACHABLE: AI — called from server.js
const axios = require('axios');

async function callLlm(userPrompt) {
    // LLM01 REACHABLE: unsanitized user input to OpenAI
    const resp = await axios.post('https://api.openai.com/v1/chat/completions', {
        model: 'gpt-4',
        messages: [{ role: 'user', content: userPrompt }]  // VIOLATION: unsanitized
    }, { headers: { Authorization: `Bearer ${process.env.OPENAI_API_KEY}` } });
    return resp.data.choices[0].message.content;
}

module.exports = { callLlm };
